File: WebDriverTests_Java/WebDriverTests/src/test/java/core/webDriver/WebDriverDownloadDirectory.java
package core.webDriver;

public final class WebDriverDownloadDirectory
{
    public static final String DirectoryPath = (new java.io.File("..\\..\\..\\.downloads")).getAbsolutePath();
//    private static final ILog logger = LogManager(WebDriverDownloadDirectory.class);

    public static void Clean()
    {
        for (String file : GetDownloadedFileNames())
        {
            (new java.io.File(file)).delete();
        }
    }

    public static String[] GetDownloadedFileNames()
    {
//        if (!(new java.io.File(DirectoryPath)).isDirectory())
//        {
            return new String[0];
//        }
//        return Directory.GetFiles(DirectoryPath).Where(name -> !name.endsWith(".crdownload") && !name.endsWith(".tmp")).ToArray();
    }


    public static void WaitDownloadFiles(int expectedCountFiles, int maximalWaitTime) throws InterruptedException {
        WaitDownloadFiles(expectedCountFiles, maximalWaitTime, 100);
    }

    public static void WaitDownloadFiles(int expectedCountFiles) throws InterruptedException {
        WaitDownloadFiles(expectedCountFiles, 15000, 100);
    }

public static void WaitDownloadFiles(int expectedCountFiles, int maximalWaitTime, int sleepInterval) throws InterruptedException {
//        logger.InfoFormat("Wait with time {0} sec. is started", maximalWaitTime / 1000);
//    logger.InfoFormat("Wait with time {0} sec. is started", maximalWaitTime / 1000);
    long startTicks = java.time.LocalDateTime.now().getSecond();
    while (true) {
        long currentTicks = java.time.LocalDateTime.now().getSecond();
        if (currentTicks - startTicks > maximalWaitTime * 10000L) {
//            int actualCountFiles = GetDownloadedFileNames().getLength();
//            if (actualCountFiles == expectedCountFiles) {
//                logger.Info(String.format("Скачано %1$s файлов, ожидается %2$s.", actualCountFiles, expectedCountFiles));
//            }
            throw new RuntimeException("Время ожидания истекло");
        }

        if (GetDownloadedFileNames().length == expectedCountFiles) {
            break;
        }
        Thread.sleep(sleepInterval);
    }
//    logger.Info("Wait finished");
}
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.Utilities/WebTestCore/TestSystem/WebDriverDownloadDirectory.cs
using System;
using System.IO;
using System.Linq;
using System.Threading;
using log4net;

namespace Autotests.Utilities.WebTestCore.TestSystem
{
    internal static class WebDriverDownloadDirectory
    {
        public static readonly string DirectoryPath = Path.GetFullPath(@"..\..\..\.downloads");
        private static readonly ILog logger = LogManager.GetLogger(typeof (WebDriverDownloadDirectory));

        public static void Clean()
        {
            foreach (string file in GetDownloadedFileNames())
                File.Delete(file);
        }

        public static string[] GetDownloadedFileNames()
        {
            if (!Directory.Exists(DirectoryPath))
                return new string[0];
            return
                Directory.GetFiles(DirectoryPath)
                    .Where(name => !name.EndsWith(".crdownload") && !name.EndsWith(".tmp"))
                    .ToArray();
        }

        public static void WaitDownloadFiles(int expectedCountFiles, int maximalWaitTime = 15000,
            int sleepInterval = 100)
        {
            logger.InfoFormat("Wait with time {0} sec. is started", maximalWaitTime/1000);
            long startTicks = DateTime.UtcNow.Ticks;
            while (true)
            {
                long currentTicks = DateTime.UtcNow.Ticks;
                if (currentTicks - startTicks > maximalWaitTime*10000L)
                {
                    int actualCountFiles = GetDownloadedFileNames().Length;
                    if (actualCountFiles == expectedCountFiles)
                        logger.Info(string.Format("Скачано {0} файлов, ожидается {1}.", actualCountFiles,
                            expectedCountFiles));
                    throw new Exception("Время ожидания истекло");
                }

                if (GetDownloadedFileNames().Length == expectedCountFiles)
                    break;
                Thread.Sleep(sleepInterval);
            }
            logger.Info("Wait finished");
        }
    }
}